File: Study03_IntentExample/app/src/main/java/com/example/study03_intentexample/MainActivity.java
package com.example.study03_intentexample;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private Button btn_move;
    private EditText et_text;
    private String str;
    ImageView iv_test;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        et_text = findViewById(R.id.et_text);


        btn_move = findViewById(R.id.btn_move);
        btn_move.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                str = et_text.getText().toString();
                Intent intent = new Intent(MainActivity.this , SubActivity.class);
                intent.putExtra("str",str);
                startActivity(intent); //액티비티 이동해주는 구문

            }
        });

        iv_test = (ImageView) findViewById(R.id.iv_test);
        iv_test.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(getApplicationContext(),"연출이잘생김",Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: yeonchulapp/app/src/main/java/com/example/yeonchulapp/MainActivity.kt
package com.example.yeonchulapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button.setOnClickListener(object : View.OnClickListener{
            override fun onClick(p0: View?) {
                // to do ..

            }

        })
        // 1. kotlin inteface 가 아닌 자바 인터페이스여야 함
        // 2. 그 인터페이스는 딱 하나의 메소드를 가져야 합니다.

        button.setOnClickListener {

        }
    }
}